eva-lib/src/main/java/uk/ac/ebi/eva/lib/repository/VariantStudySummaryRepositoryImpl.java

```diff
@@ -0,0 +1,141 @@
+import org.springframework.data.mongodb.repository.MongoRepository;
+
+import uk.ac.ebi.eva.commons.models.data.VariantSourceEntity;
+import uk.ac.ebi.eva.lib.repository.projections.VariantStudySummary;
+
+import java.util.List;
+
+import static org.springframework.data.mongodb.core.aggregation.Aggregation.group;
+import static org.springframework.data.mongodb.core.aggregation.Aggregation.match;
+import static org.springframework.data.mongodb.core.aggregation.Aggregation.project;
+
+/**
+ * Implementation of the {@link VariantStudySummaryRepositoryCustom} interface (relationship with
+ * {@link VariantStudySummaryRepository} inferred by Spring).
+ * <p>
+ * The repository implementation could not be made implicitly by spring, due to use of MongoDB Aggregation Framework.
+ */
+public class VariantStudySummaryRepositoryImpl implements VariantStudySummaryRepositoryCustom {
+
+    private static final String STUDY_ID = "studyId";
+
+    private static final String STUDY_NAME = "studyName";
+
+    private static final String FILES_COUNT = "filesCount";
+
+    private static final String ID = "_id.";
+
+    private MongoTemplate mongoTemplate;
+
+    public VariantStudySummaryRepositoryImpl(MongoDbFactory mongoDbFactory, MappingMongoConverter mappingMongoConverter) {
+        mongoTemplate = new MongoTemplate(mongoDbFactory, mappingMongoConverter);
+    }
+
+    /**
+     * the equivalent intended query is:
+     * db.files.aggregate([
+     *  {$match : { $or:[{"sid" : "studyNameOrId"} , {"sname": "studyNameOrId"}]} },
+     *  {$group:{_id: {studyId:"$sid",studyName:"$sname"}, filesCount:{$sum:1}}},
+     *  {$project:{"studyId" : "$_id.studyId", "studyName" : "$_id.studyName", "_id" : 0, "filesCount":"$filesCount" }}
+     *  ])
+     * First, match only the documents that contain the argument as studyName or studyId.
+     * Then, group by both "studyName" and "studyId", and count the documents in the group into an accumulator field
+     * "filesCount". The idea behind grouping by 2 fields is not to have the combinations (because a studyId will have
+     * always the same studyName), but to include both fields without further projections or queries.
+     * Then, flatten the document from {_id: {sid, sname}, filesCount} to {sid, sname, filesCount} which is ready to
+     * be mapped into a {@link VariantStudySummary}.
+     */
+    public VariantStudySummary findByStudyNameOrStudyId(String studyNameOrId) {
+        Aggregation aggregation = Aggregation.newAggregation(
+                matchByNameOrId(studyNameOrId),
+                groupAndCount(),
+                projectAndFlatten()
+        );
+
+        AggregationResults<VariantStudySummary> studies = mongoTemplate.aggregate(aggregation,
+                VariantSourceEntity.class,
+                VariantStudySummary.class);
+
+        VariantStudySummary variantStudySummary;
+        if (studies.getMappedResults().size() == 0) {
+            variantStudySummary = null;
+        } else {
```

Comment:
if a document in the files collection has the same studyId as another document, but has a different study name, yes, it will happen. This would be a DB inconsistency and I don't really know what should we do here. any thoughts @cyenyxe ?